sdk/docs/pages/dapp-kit/wallet-hooks/useSignAndExecuteTransactionBlock.mdx

```diff
@@ -0,0 +1,70 @@
+import { UseSignAndExecuteTransactionBlockExample } from '../../../examples/wallet-hooks';
+
+# `useSignAndExecuteTransactionBlock`
+
+The `useSignAndExecuteTransactionBlock` hook can be used to prompt the user to sign and execute a
+transaction block with their wallet
+
+```ts
+import {
+	ConnectButton,
+	useCurrentAccount,
+	useSignAndExecuteTransactionBlock,
+} from '@mysten/dapp-kit';
+import { useState } from 'react';
+
+function MyComponent() {
+	const { mutate: signAndExecuteTransactionBlock } = useSignAndExecuteTransactionBlock();
+	const [digest, setDigest] = useState('');
+	const currentAccount = useCurrentAccount();
+
+	return (
+		<div style={{ padding: 20 }}>
+			<ConnectButton />
+			{currentAccount && (
+				<>
+					<div>
+						<button
+							onClick={() => {
+								signAndExecuteTransactionBlock(
+									{
+										transactionBlock: new TransactionBlock(),
+										chain: 'sui:devnet',
+									},
+									{
+										onSuccess: (result) => {
+											console.log('executed transaction block', result);
+											setDigest(result.digest);
+										},
+									},
+								);
+							}}
+						>
+							Sign and execute transaction block
+						</button>
+					</div>
+					<div>Digest: {digest}</div>
+				</>
+			)}
+		</div>
+	);
+}
+```
+
+## Example
+
+<UseSignAndExecuteTransactionBlockExample />
+
+## Arguments
+
+- `transactionBlock`: The transaction block to sign
+- `chain`: The chain identifier to sign the transaction block should be signed for
+- `requestType`: `WaitForEffectsCert` or `WaitForLocalExecution`. Determines when the RPC node
+  should return the response. Default to be `WaitForLocalExecution`
+- `options`:
```

Comment:
Can we link to this method in the SDK so that we don't need to duplicate the options definitions here?